searches/interpolation_search.py

```diff
@@ -3,7 +3,7 @@
 """
 
 
-def interpolation_search(sorted_collection, item):
+def interpolation_search(sorted_collection: list[int], item: int) -> int | None:
     """Pure implementation of interpolation search algorithm in Python
     Be careful collection must be ascending sorted, otherwise result will be
```

Comment:
Please test for this condition.  Give it an unsorted input and lets see if it fails or not.

Reply:
An unsorted input doesn't work with this implementation unfortunately. For example ```>>> interpolation_search([5, 0, 1], 2) == None``` will return ```True``` despite 2 not being in the array. I wasn't sure how to handle this as there is a helper function called ```__assert_sorted```. I considered adding that code at the beginning of each function but that changes the time complexity of the functions from logn to nlogn. If something is done to sanitize the input I assume it should be done in a separate PR.